3-networks/modules/restricted_shared_vpc/main.tf

```diff
@@ -14,16 +14,32 @@
  * limitations under the License.
  */
 
-/******************************************
-  Shared VPC configuration
- *****************************************/
-
 locals {
-  vpc_name                   = "${var.environment_code}-shared-restricted"
+  mode                       = var.mode == null ? "" : var.mode == "hub" ? "-hub" : "-spoke"
+  vpc_name                   = "${var.environment_code}-shared-restricted${local.mode}"
   network_name               = "vpc-${local.vpc_name}"
   restricted_googleapis_cidr = "199.36.153.4/30"
 }
 
+/******************************************
+  Restricted Network Hub
+*****************************************/
+
+data "google_projects" "restricted_net_hub" {
+  for_each = var.mode == "spoke" ? toset(["yes"]) : toset([])
+  filter   = "parent.id:${split("/", data.google_active_folder.common.name)[1]} labels.application_name=org-restricted-net-hub lifecycleState=ACTIVE"
+}
+
+data "google_compute_network" "vpc_restricted_net_hub" {
+  for_each = var.mode == "spoke" ? toset(["yes"]) : toset([])
```

Comment:
You are right, for if-else statements count is slightly better. I'm just trying to use for_each instead of count whenever possible. I will replace with counts.